spec/javascripts/unit/modules/cookie-settings.spec.js

```diff
@@ -0,0 +1,176 @@
+"use strict";
+describe('cookieSettings', function() {
+  var cookieSettings,
+      container,
+      element,
+      confirmationContainer,
+      fakePreviousURL;
+
+  beforeEach(function() {
+    cookieSettings = new GOVUK.Modules.CookieSettings()
+    GOVUK.Modules.CookieSettings.prototype.getReferrerLink = function () {
+      return fakePreviousURL
+    }
+
+    container = document.createElement('div')
+    container.innerHTML =
+      '<form data-module="cookie-settings">' +
+        '<input type="radio" id="settings-on" name="cookies-settings" value="on">' +
+        '<input type="radio" id="settings-off" name="cookies-settings" value="off">' +
+        '<input type="radio" name="cookies-usage" value="on">' +
+        '<input type="radio" name="cookies-usage" value="off">' +
+        '<input type="radio" name="cookies-campaigns" value="on">' +
+        '<input type="radio" name="cookies-campaigns" value="off">' +
+        '<button id="submit-button" type="submit">Submit</button>' +
+      '</form>'
+
+    document.body.appendChild(container)
+    element = document.querySelector('[data-module=cookie-settings]')
+  });
+
+  afterEach(function() {
+    document.body.removeChild(container)
+  });
+
+  describe('setInitialFormValues', function () {
+    it('sets a consent cookie by default', function() {
+      GOVUK.cookie('cookie_policy', null)
+
+      spyOn(window.GOVUK, 'setDefaultConsentCookie').and.callThrough()
+      cookieSettings.start(element)
+
+      expect(window.GOVUK.setDefaultConsentCookie).toHaveBeenCalled()
+    });
+
+    it('sets all radio buttons to the default values', function() {
+      cookieSettings.start(element)
+
+      var radioButtons = element.querySelectorAll('input[value=on]')
+      var consentCookieJSON = JSON.parse(window.GOVUK.cookie('cookie_policy'))
+
+      for(var i = 0; i < radioButtons.length; i++) {
+        var name = radioButtons[i].name.replace('cookies-', '')
+
+        if (consentCookieJSON[name]) {
+          expect(radioButtons[i].checked).toBeTruthy()
+        } else {
+          expect(radioButtons[i].checked).not.toBeTruthy()
+        }
+      }
+    });
+  });
+
+  describe('submitSettingsForm', function() {
+    beforeEach(function() {
+      confirmationContainer = document.createElement('div')
```

Comment:
We should be able to DRY these up from here and `showConfirmationMessage` into the parent `beforeEach`